client/src/components/userProvider/index.js

```diff
@@ -1,11 +1,30 @@
-import React, { useState, createContext } from 'react';
+import React, { useState, createContext, useEffect } from 'react';
 import PropTypes from 'prop-types';
+import axios from 'axios';
 
 const { node } = PropTypes;
 export const context = createContext();
 
 function userProvider({ children }) {
   const isAuthenticated = useState(false);
+  const [, setIsAuth] = isAuthenticated;
+
+  useEffect(() => {
+    const { CancelToken } = axios;
+    const source = CancelToken.source();
+    const request = async () => {
```

Comment:
I don’t think you have to put this inside a function and call it. The useEffect itself can be declared async and then you can remove this wrapping function (I think).